backend/api/controllers/PullRequestController.js

```diff
@@ -1,20 +1,56 @@
 const express = require('express')
 const PullRequest = require('../../services/PullRequest')
+const redisClient = require('../../redis')
 
 const PullRequestController = express()
 
 PullRequestController.get('/:year?', async function (req, res) {
   let response = {}
-  let pullRequest = new PullRequest(req.params.year)
+  let year = req.params.year
+  let pullRequest = new PullRequest(year)
 
   try {
-    response = await pullRequest.getAll()
+    const users = await redisClient.exists(`users:${year}`)
 
-    response
-      .groupByUser()
-      .sortByMostActive()
+    if (users === 0) {
+      response = await pullRequest.getAll()
 
-    res.send(response.data)
+      response
+        .groupByUser()
+        .sortByMostActive()
+
+      let arrayOfOrderedUsers = []
+      let data = response.data
+      let promises = []
+
+      for (let i = 0; i < data.length; i++) {
+        promises[i] = redisClient.sadd(`pull-requests:${year}:${data[i].username}`,
+          data[i].pullRequests)
+        arrayOfOrderedUsers.push(data[i].pullRequests.length, data[i].username)
```

Comment:
The `lengh` must be, instead, the total of already stored pull requests, read from redis, because we will be adding to the pull requests set.

We can use `SMEMBERS` to read the pull requests.

Reply:
Also, lets make it more semantic:

```js
const score = data[i].pullRequests.length
const username = data[i].username
arrayOfOrderedUsers.push(score, username)
```

Reply:
You can't use `const` in a `for..loop` only in `for..in..loop`. I will just make them variables.

Reply:
> The `lengh` must be, instead, the total of already stored pull requests, read from redis, because we will be adding to the pull requests set.
> 


Why fetch from redis if we already know that `users === 0`? The above piece of code is placed inside a condition which only runs the code if there are no users (therefore, no PRs) in redis.

Reply:
> You can't use const in a for..loop only in for..in..loop. I will just make them variables.

One of the reasons I prefer to use `.forEach` :sweat_smile: 
But variables will work as well. :)

Reply:
> Why fetch from redis if we already know that users === 0? The above piece of code is placed inside a condition which only runs the code if there are no users (therefore, no PRs) in redis.

I was thinking about when we update the db, adding more PRs for existing users.
Makes sense to let it for later in #7, although. :+1: